swagger.js

```diff
@@ -31,6 +32,104 @@ const doc = {
         },
       },
     },
+    UserWithAlbums: {
+      $id: '64d6ca92a15d2e18ab96a2a3',
+      $name: 'John Doe',
+      $username: 'johndoe',
+      $email: 'john@example.com',
+      $role: { '@enum': ['user', 'admin'] },
+      profileImage: {
+        url: 'https://example.com/profile-image.jpg',
+      },
+      creditCardInfo: {
+        hashedNumber: '1234567890',
+        expiry: '12/24',
+        preferredPaymentOption: {
+          '@enum': ['credit card', 'paypal', 'google pay'],
+        },
+      },
+      $purchasedAlbums: [
+        {
+          $_id: '64ef50c8c5551074444547bc',
+          $user: '64d6ca92a15d2e18ab96a2a3',
+          $album: {
+            $id: '64d2a94c793389a43fc5a8d6',
+            $albumName: '...Baby one more time',
+            $artistName: 'Britney Spears',
+            $createdAt: '2023-08-08T20:45:00.942Z',
+            $updatedAt: '2023-08-17T10:10:42.861Z',
+            $price: 20.99,
+            image:
+              'https://i.scdn.co/image/ab67616d0000b27371cae34ad5a39bdab78af13e',
+            spotifyUrl:
+              'https://api.spotify.com/v1/albums/6r1lh7fHMB499vGKtIyJLy',
+            releaseDate: '2023-07-28T00:00:00.000Z',
+            averageRating: 0,
+            numOfReviews: 0,
+          },
+        },
+      ],
+    },
+    Album: {
+      $id: '64d2ae2308a725b72bd5c0cf',
+      $albumName: 'The Dark Side of the Moon',
+      $artistName: 'Pink Floyd',
+      price: 9.99,
+      image: '/uploads/example.jpg',
+      releaseDate: '2022-03-01T00:00:00.000Z',
+      category: 'rock',
+      $spotifyUrl: 'https://open.spotify.com/album/4LH4d3cOWNNsVw41Gqt2kv',
+      averageRating: 4.5,
+      numOfReviews: 2,
+    },
+    NewAlbum: {
+      $id: '64d2ae2308a725b72bd5c0cf',
+      $albumName: 'The Dark Side of the Moon',
+      $artistName: 'Pink Floyd',
+      price: 9.99,
+      image: '/uploads/example.jpg',
+      releaseDate: '2022-03-01T00:00:00.000Z',
+      category: 'rock',
+      $spotifyUrl: 'https://open.spotify.com/album/4LH4d3cOWNNsVw41Gqt2kv',
+    },
+    AlbumsWithUsers: {
```

Comment:
In the albumController file you only refer to `AlbumWithUsers` (no `'s'` on the Album). Since swagger-autogen couldn't find a matching definition in this file, it just defaults to "string" in the docs:

![](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/9094098/62188159-e9dd-4ee8-8dea-81a5628384f8)

You're using `#/definitions/AlbumWithUsers` in the `getAlbumWithAllUsersWhoPurchasedIt`, which returns an object like:

```
{
  "album": {
    "_id": "64d2a94c793389a43fc5a8d6",
    "artistName": "F",
    "albumName": "AUSTIN",
    "price": 0,
    "image": "https://i.scdn.co/image/ab67616d0000b27371cae34ad5a39bdab78af13e",
    "releaseDate": "2023-07-28T00:00:00.000Z",
    "spotifyUrl": "https://api.spotify.com/v1/albums/6r1lh7fHMB499vGKtIyJLy",
    "averageRating": 0,
    "numOfReviews": 0,
    "__v": 0,
    "createdAt": "2023-08-08T20:45:00.942Z",
    "updatedAt": "2023-08-17T10:10:42.861Z",
    "purchasedByUsers": [
      {
        "_id": "64ef50c8c5551074444547bc",
        "album": "64d2a94c793389a43fc5a8d6",
        "user": {
          "_id": "64d6ca92a15d2e18ab96a2a3",
          "name": "Akos123123",
          "email": "akos123@example.com",
          "role": "user",
          "__v": 0,
          "username": "akos92",
          "id": "64d6ca92a15d2e18ab96a2a3"
        },
        "__v": 0
      }
    ],
    "id": "64d2a94c793389a43fc5a8d6"
  },
  "purchasingUsersCount": 1
}
```

But this code here seems to be defining a _user_ with their list of _purchased albums_. I think that's what we had in the user docs for UserWithAlbums. But this should kind of be the inverse of that.

Reply:
@akosasante thank you!!!  will commit it now, just to make sure- to write down the definition correctly- we should first make a request to the url and then convertin to js format and check in models (User and Album) which fields are required and put a $ sign ?